tests/unit_tests/CMakeLists.txt

```diff
@@ -66,6 +66,11 @@ set(dft_TEST_LIST
 set(dft_TEST_LINK "")
 
 foreach(domain ${TARGET_DOMAINS})
+  # TODO: Add sparse_blas tests and CMake logic
+  if (domain STREQUAL "sparse_blas")
+    continue()
```

Comment:
Should we have at least MKLCPU enabled and tested before merging this implementation?

Reply:
We discussed how we wanted to split the sparse PRs but you weren't included in these discussions I think, sorry about that.
In general we found that it's best to aim for smaller PRs so we've agreed that the first PR would throw unsupported for every sparse function. The tests are quite large as we need to add a naive reference operation for all the sparse operations. We're planning to add support for each operation in separate PRs along with their own tests.

Reply:
I agree with @mkrainiuk. Before enabling any backend, we need to have tests implemented to cover these backend, even though they throw unsupported exceptions. Without tests, how are you validating if the backend interfaces are implemented correctly?
Also, the title of the PR is misleading. A user reading the title might think, the sparse MKLCPU backend is fully implemented. Can you please update the PR name to clarify that? Unfortunately, not everybody reads the PR descriptions.

Reply:
> Without tests, how are you validating if the backend interfaces are implemented correctly?

Well that is the challenge when we are aiming for small PRs. The way I see it the only way to check that the MKLCPU backend is implemented correctly would be to have all of the tests and reference implementation in the first PR. That would be quite big for the sparse domain.

> Before enabling any backend, we need to have tests implemented to cover these backend, even though they throw unsupported exceptions.

So are you suggesting to add tests that check that every operation throws unsupported for now? I think that would be misleading for users as it would look like all sparse tests are passing but I can do that.

I've updated the PR title.

Reply:
I should add that the logic added here is supposed to be as simple as possible: it is only adding the prototypes from the specification. In theory it would not need to be changed in future PRs unless the spec changes.
The best test I could think of for this PR is what I have written in the PR description i.e. the runtime and compile-time dispatch examples are building and linking without any undefined references so the global structure should be correct.